### DISCUSSION TITLE
SWR with React Router 6.4. data loading

### DISCUSSION THREAD

**Discussion by hovsater** [vercel/swr#2237] [Q&A] [answered]
Have anyone successfully used SWR with React Router 6.4 and its new [data loading](https://reactrouter.com/en/main/start/overview#data-loading) capabilities? While there's an official [example](https://reactrouter.com/en/main/guides/data-libs) of using React Router with [React Query](https://react-query-v3.tanstack.com/), I'd love to see how it would be done using SWR. Since data loading is happening outside of the React render lifecycle, hooks aren't allowed and I was unsure how that would translate into the current SWR API.

Any pointers would be much appreciated. 🙂

**Comment by jimmyjansen93** [comment]
Came here to ask the same question. Very interested in making this happen as I could then preload data based on the routes

**Comment by fouteox** [comment]
Hi, try this repository : https://github.com/osmellodgify/swr-react-router

**Comment by promer94** [comment, marked as answer]
https://beta-swr.vercel.app/docs/prefetching#programmatically-prefetch

we have a new preload api which could be used for react-router
